subsys/net/ip/net_context.c

```diff
@@ -2471,6 +2471,12 @@ static int context_sendto(struct net_context *context,
 					   (iface = net_if_get_by_index(
 						   context->options.ipv6_mcast_ifindex)));
 			}
+			if (net_ipv6_is_ll_addr(&addr6->sin6_addr) &&
```

Comment:
Shouldn't `sin6_scope_id` be handled for any context type, why is it limited to UDP only?